niworkflows/interfaces/images.py

```diff
@@ -305,18 +305,24 @@ def _run_interface(self, runtime):
             xyz_unit = "mm"
 
         # Set a 0.05mm threshold to performing rescaling
-        atol = {"meter": 1e-5, "mm": 0.01, "micron": 10}[xyz_unit]
+        atol = {"meter": 5e-5, "mm": 0.05, "micron": 50}[xyz_unit]
+        # if 0.01 > difference > 0.001mm, freesurfer won't be able to merge the images
+        atol_fine = {"meter": 1e-6, "mm": 0.001, "micron": 1}[xyz_unit]
 
         # Rescale => change zooms
+        # Fix zooms => set zooms without changing affine
```

Comment:
Does this make sense?

Reply:
> I'm not sure `set_zooms()` is the appropriate implementation, since that won't update the sform. I think what we probably want to do is just unconditionally set the affine even if we don't resample the dataobj. This may have knock-on effects into how we implement the other rows.

What about generating a new affine extracting the direction cosines of the old one and composing the directions with the new zooms?